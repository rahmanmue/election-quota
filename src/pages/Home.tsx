import React from "react";
import Navbar from "@/components/landing-page/Navbar";
import Footer from "@/components/landing-page/Footer";
import Hero from "@/components/landing-page/Hero";
import About from "@/components/landing-page/About";
import ScrollToTop from "@/components/ScrollToTop";
import Calculation from "@/components/landing-page/Calculation";

const Home: React.FC = () => {
  return (
    <>
      <Navbar />
      <Hero />
      <Calculation />
      <About />
      <Footer />
      <ScrollToTop />
      <div className="shadow-hero" />
    </>
  );
};

export default Home;
