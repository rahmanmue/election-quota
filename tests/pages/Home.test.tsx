import React from "react";
import { render, screen } from "@testing-library/react";
import { vi } from "vitest";
import Home from "../../src/pages/Home"; // Sesuaikan path ke file Home

// Mock komponen child untuk testing
vi.mock('@/components/landing-page/Navbar', () => ({
  default: () => <div>Navbar Component</div>,
}));
vi.mock('@/components/landing-page/Hero', () => ({
  default: () => <div>Hero Component</div>,
}));
vi.mock('@/components/landing-page/Calculation', () => ({
  default: () => <div>Calculation Component</div>,
}));
vi.mock('@/components/landing-page/About', () => ({
  default: () => <div>About Component</div>,
}));
vi.mock('@/components/landing-page/Footer', () => ({
  default: () => <div>Footer Component</div>,
}));
vi.mock('@/components/ScrollToTop', () => ({
  default: () => <div>ScrollToTop Component</div>,
}));

describe("Home Component", () => {
  it("renders all child components", () => {
    render(<Home />);

    // Memastikan semua komponen muncul
    expect(screen.getByText("Navbar Component")).toBeInTheDocument();
    expect(screen.getByText("Hero Component")).toBeInTheDocument();
    expect(screen.getByText("Calculation Component")).toBeInTheDocument();
    expect(screen.getByText("About Component")).toBeInTheDocument();
    expect(screen.getByText("Footer Component")).toBeInTheDocument();
    expect(screen.getByText("ScrollToTop Component")).toBeInTheDocument();
  });

  it("calls window.scrollTo on mount", () => {
    const scrollToMock = vi.fn();
    // Mock window.scrollTo
    window.scrollTo = scrollToMock;

    render(<Home />);

    // Memastikan scrollTo dipanggil
    expect(scrollToMock).toHaveBeenCalledWith(0, 0);
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
