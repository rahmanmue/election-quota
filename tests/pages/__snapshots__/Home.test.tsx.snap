// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    Navbar Component
  </div>
  <div>
    Hero Component
  </div>
  <div>
    Calculation Component
  </div>
  <div>
    About Component
  </div>
  <div>
    Footer Component
  </div>
  <div>
    ScrollToTop Component
  </div>
  <div
    class="shadow-hero"
  />
</DocumentFragment>
`;
